gpkit/constraints/set.py

```diff
@@ -168,6 +168,8 @@ def subinplace(self, subs):
         "Substitutes in place."
         for constraint in self:
             constraint.subinplace(subs)
+        for key in set(self.substitutions).intersection(subs):
+            self.substitutions[subs[key]] = self.substitutions[key]
```

Comment:
I could be wrong here, but something seems fishy here:

`subs[key]` is a value.
... and now we're using that value as a key into `self.substitutions`?

just making sure.